mypy_json_report.py

```diff
@@ -120,8 +120,13 @@ def _parse_command(args: argparse.Namespace) -> None:
         processors.append(tracker)
 
     messages = MypyMessage.from_lines(sys.stdin)
+
+    # Sort the lines by the filename otherwise itertools.groupby() will make
+    # multiple groups for the same file name if the lines are out of order.
+    messages_sorted = sorted(list(messages), key=operator.attrgetter("filename"))
```

Comment:
**Question (non-blocking):** Is the `list` required here?

Reply:
Yes, because `MypyMessage.from_lines` uses `yield` instead of returning a list. `sorted()` can't work on iterators. We could potentially change `from_lines` to instead return a list, but it's probably good that it doesn't impose the memory usage hit unnecessarily.

Reply:
Sorry, I was wrong. I just tried to remove it now and it works fine without it. I must have added it as part of my work on this and forgotten to remove it again.